src/index.ts

```diff
@@ -1,84 +1,90 @@
+const isValidURL = (str: string) => URL_LONG.test(str) || URL_SHORT.test(str);
 
-export function createStyle (doc: Document, cssText: string) {
-  const style: HTMLStyleElement = doc.createElement('style')
+export function createStyle(doc: Document, cssText: string) {
+  const style: HTMLStyleElement = doc.createElement("style");
 
-  style.type = 'text/css'
-  style.appendChild(window.document.createTextNode(cssText))
+  style.type = "text/css";
+  style.appendChild(window.document.createTextNode(cssText));
 
-  return style
+  return style;
 }
 
-export function createLinkStyle (doc: Document, url: string) {
-  const style: HTMLLinkElement = doc.createElement('link')
+export function createLinkStyle(doc: Document, url: string) {
+  const style: HTMLLinkElement = doc.createElement("link");
 
-  style.type = 'text/css'
-  style.rel = 'stylesheet'
-  style.href = url
+  style.type = "text/css";
+  style.rel = "stylesheet";
+  style.href = url;
 
-  return style
+  return style;
 }
 
-export function createIFrame (parent: HTMLElement) {
-  const el: HTMLIFrameElement = window.document.createElement('iframe')
-  const css = 'visibility:hidden;width:0;height:0;position:absolute;z-index:-9999;bottom:0;'
+export function createIFrame(parent: HTMLElement) {
+  if (typeof window !== "undefined") {
+    const el: HTMLIFrameElement = window.document.createElement("iframe");
+    const css =
+      "visibility:hidden;width:0;height:0;position:absolute;z-index:-9999;bottom:0;";
 
-  el.setAttribute('src', 'about:blank')
-  el.setAttribute('style', css)
-  el.setAttribute('width', '0')
-  el.setAttribute('height', '0')
-  el.setAttribute('wmode', 'opaque')
+    el.setAttribute("src", "about:blank");
+    el.setAttribute("style", css);
+    el.setAttribute("width", "0");
+    el.setAttribute("height", "0");
+    el.setAttribute("wmode", "opaque");
 
-  parent.appendChild(el)
+    parent.appendChild(el);
 
-  return el
+    return el;
+  }
+  return null;
 }
 
 export interface PrintdOptions {
   /** Parent element where the printable element will be appended. */
-  parent?: HTMLElement
+  parent?: HTMLElement;
   /** Specifies a custom document head elements */
-  headElements?: HTMLElement[]
+  headElements?: HTMLElement[];
   /** Specifies a custom document body elements */
-  bodyElements?: HTMLElement[]
+  bodyElements?: HTMLElement[];
 }
 
 export interface PrintdCallbackArgs {
   /** Iframe reference */
-  iframe: HTMLIFrameElement
+  iframe: HTMLIFrameElement;
   /** HTMLElement copy reference */
-  element?: HTMLElement
+  element?: HTMLElement;
   /** Function to launch the print dialog after content was loaded */
-  launchPrint: Function
+  launchPrint: Function;
 }
 
-export type PrintdCallback = (args: PrintdCallbackArgs) => void
+export type PrintdCallback = (args: PrintdCallbackArgs) => void;
 
 const DEFAULT_OPTIONS: PrintdOptions = {
-  parent: window.document.body,
+  parent: typeof window !== "undefined" ? window.document.body : null,
```

Comment:
**Third**, As I understood, do you want to check for the global `window` object, right?
If it's `null` then `this.parent` will be `null` as well but I don't see again anywhere if you are covering that case.
Please fix that and add the necessary unit tests for.

Reply:
there is an if statement ... if the typeof window is undefined then only will be the body is null otherwise its will be window.document.body

Reply:
That's something obvious, it's no problem with checking for global `window` value but the problem is that `parent` will receive eventually two different kind of values. So I don't see any concrete handing in the rest of the code.